src/elm/AstarteApi.elm

```diff
@@ -609,10 +610,17 @@ groupList apiConfig resultMsg =
 
 addDeviceToGroup : Config -> String -> String -> (Result Error () -> msg) -> Cmd msg
 addDeviceToGroup apiConfig groupName deviceId resultMsg =
+    let
+        {- Double encoding to preserve the URL format when groupName contains % and / -}
+        encodedGroupName =
+            groupName
+            |> Url.percentEncode
+            |> Url.percentEncode
+    in
     Http.request
         { method = "POST"
         , headers = buildHeaders apiConfig.token
-        , url = buildUrl apiConfig.secureConnection apiConfig.appengineUrl [ "v1", apiConfig.realm, "groups", groupName, "devices" ] []
+        , url = buildUrl apiConfig.secureConnection apiConfig.appengineUrl [ "v1", apiConfig.realm, "groups", encodedGroupName, "devices" ] []
```

Comment:
Do you think it could be useful to add a test for this `addDeviceToGroup` API too?